core/store/src/lib.rs

```diff
@@ -222,6 +224,27 @@ impl StoreUpdate {
         }
         self.storage.write(self.transaction).map_err(|e| e.into())
     }
+
+    fn inc_gc(&mut self, column: DBCol) {
+        let new_count: u64 = if let Some(count) = self.gc_count.get(&(column as u64)) {
+            count + 1
+        } else if let Ok(Some(count)) = self.storage.get(
+            DBCol::ColGCCount,
+            &borsh::ser::BorshSerialize::try_to_vec(&(column as u64)).unwrap(),
+        ) {
+            let count: u64 = borsh::de::BorshDeserialize::try_from_slice(&count).unwrap();
+            count + 1
+        } else {
+            1
+        };
+        self.set_ser(
+            DBCol::ColGCCount,
+            &borsh::ser::BorshSerialize::try_to_vec(&(column as u64)).unwrap(),
+            &new_count,
+        )
+        .unwrap();
```

Comment:
> Call gc method on all gcing now. test failed due to:
> 
> ```
> debug_assert!(
>             self.transaction.ops.len()
>                 == self
>                     .transaction
>                     .ops
>                     .iter()
>                     .map(|op| match op {
>                         DBOp::Insert { col, key, .. } => {
>                             (*col as u8, key)
>                         }
>                         DBOp::Delete { col, key } => {
>                             (*col as u8, key)
>                         }
>                     })
>                     .collect::<std::collections::HashSet<_>>()
>                     .len(),
>             "Transaction overwrites itself: {:?}",
>             self
>         );
> ```
> 
> Trying to figure out a way to write "update times of gc on col during this batch of update" only once, to pass this assert

You can't do `set_ser` here. The only place where you can do `set_ser` is `ChainStoreUpdate::finalize`.

Reply:
Ah yeah, the finalized func is a good place to set all column gc count